File: src/main/kotlin/dev/jefy/connectpro/user/applicaion/dtos/CreateAccountRequest.java
package dev.jefy.connectpro.user.applicaion.dtos;


import org.jspecify.annotations.NullMarked;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;

/**
 * @author Jôph Yamba
 */

@NullMarked
public record CreateAccountRequest(
        @NotBlank(message = "name cannot be empty")
        String name,
        @Email(message = "email is not valid email address")
        @NotBlank(message = "email cannot be empty")
        String email,
        @NotBlank(message = "password cannot be empty")
        String password,
        @NotBlank(message = "confirm value cannot be empty")
        String confirmPassword
) {}

File: src/main/kotlin/dev/jefy/connectpro/user/application/dtos/CreateAccountRequest.kt
package dev.jefy.connectpro.user.application.dtos

import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank
import org.jspecify.annotations.NullMarked

@NullMarked
data class CreateAccountRequest(
    @field:NotBlank(message = "name cannot be empty")
    val name: String,
    @field:Email(message = "email is not valid email address")
    @field:NotBlank(message = "email cannot be empty")
    val email: String,
    @field:NotBlank(message = "password cannot be empty")
    val password: String,
    @field:NotBlank(message = "confirm value cannot be empty")
    val confirmPassword: String
)